apps/docs/build.md

```diff
@@ -172,49 +172,75 @@ different configurations get run. Here is the approximate flow of commands/files
 
    The rails asset pipeline can be configured to do bundling and minification,
    but it is a significantly less powerful tool than webpack. As a result, we
-   only use the rails asset pipeline to add "digests" to the filenames. In
-   effect, the `assets:precompile` rake task just copies over all the files that
-   were generated by webpack, adding a unique digest hash to the
-   filenames, and constructing a manifest file with the mapping. You can find
-   all these files in `dashboard/public/assets/`.
-
-   When loading a bundle with a script tag inside an erb or haml
-   file, rails will automatically add the unique digest to the url by looking at
-   the manifest file generated during precompilation. The manifest file is
-   stored at
+   only use the rails asset pipeline to add "digests" to the few assets which
+   do not already contain hashes generated by webpack, and to retain each
+   hashed/digested asset for a few deploys so that links to them do not break
+   immediately after each deploy.
```

Comment:
Two questions about this line:

1. What are the 'few assets which do not already contain hashes generated by webpack'? (I see `application.js` / `application.css` are listed in the paragraph below, but also 'a few other'.) Are these a small handful of 'legacy' assets that are in the process of being migrated over fully (so that this number will eventually be zero), or will there always be a 'few' assets in this state moving forward? Are they a certain category of asset that can be specified more specifically? (Edit: I see this is described in a bit more detail in the 'Future work' in the design doc- perhaps worth referencing that here, but not necessary, at least I understand it a bit better.)

2. Does this line mean that _all_ hashed/digested assets are retained for a few deploys by the rails asset pipeline, or _only_ the 'few assets' will be retained?

Reply:
1. The "few assets which do not already contain hashes generated by webpack" are the union of:
  a. The 85 calls to `javascript_include_tag`, `stylesheet_link_tag` or `asset_path` which do not point to `application.js` or `application.css` (as described in Future work). 
  b. `application.js` and `application.css`

(a) will be relatively easy to move into webpack.

(b) will be hard to change any time soon, so I'm advocating that we leave `application.js` and `application.css` as the only two things not being referenced via `webpack_asset_path` once follow-up work is complete.

2.  _all_ hashed/digested assets are retained for a few deploys by the rails asset pipeline.